gcc/rust/typecheck/rust-hir-path-probe.h

```diff
@@ -426,6 +456,20 @@ class PathProbeType : public TypeCheckBase, public HIR::HIRImplVisitor
     return receiver_is_type_param || receiver_is_dyn;
   }
 
+  // we should probably move to a set but not sure how to enforce the
+  // unique-ness based on the DefId yet on std::set
+  void append_candidate (PathProbeCandidate &&candidate)
+  {
+    for (const auto &c : candidates)
+      {
+	bool found = c.get_defid () == candidate.get_defid ();
+	if (found)
+	  return;
```

Comment:
I think this already assumes that `DefId`s are unique, so we should be able to use a set (unless I'm misunderstanding the above comment?)

Reply:
Yeah, i just need to figure out how to do that with std::set to say take this type but use this as the key

Reply:
I've now converted to std::set just checking the CI passes ok